sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py

```diff
@@ -0,0 +1,251 @@
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str,  # type: str
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> SipRoutingClient
+        """Factory method for creating client from connection string.
+
+        :param str conn_str: Connection string containing endpoint and credentials.
+        :returns: The newly created client.
+        :rtype: ~azure.communication.siprouting.SipRoutingClient
+        """
+        endpoint, credential = parse_connection_str(conn_str)
+        return cls(endpoint, credential, **kwargs)
+
+    @distributed_trace
+    def get_trunk(
+        self,
+        trunk_fqdn,  # type: str
+        **kwargs  # type: Any
+    ):  # type: (...) -> SipTrunk
+        """Getter for single SIP trunk.
+
+        :param trunk_fqdn: FQDN of the desired SIP trunk.
+        :type trunk_fqdn: str
+        :returns: SIP trunk with specified trunk_fqdn.
+        :rtype: ~azure.communication.siprouting.models.SipTrunk
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError, LookupError
+        """
+        if trunk_fqdn is None:
+            raise ValueError("Parameter 'trunk_fqdn' must not be None.")
+
+        config = self._rest_service.get_sip_configuration(
+            **kwargs)
+        trunk = config.trunks[trunk_fqdn]
+
+        if not trunk:
+            raise LookupError("No entry found for FQDN:" + trunk_fqdn)
+
+        return SipTrunk(fqdn=trunk_fqdn,sip_signaling_port=trunk.sip_signaling_port)
+
+    @distributed_trace
+    def set_trunk(
+        self,
+        trunk,  # type: SipTrunk
+        **kwargs  # type: Any
+    ):  # type: (...) -> None
+        """Modifies SIP trunk with the given FQDN. If it doesn't exist, adds a new trunk.
+
+        :param trunk: Trunk object to be set.
+        :type trunk: ~azure.communication.siprouting.models.SipTrunk
+        :returns: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+        """
+        if trunk is None:
+            raise ValueError("Parameter 'trunk' must not be None.")
+
+        self._patch_trunks_([trunk],**kwargs)
+
+    @distributed_trace
+    def delete_trunk(
+        self,
+        trunk_fqdn,  # type: str
+        **kwargs  # type: Any
+    ):  # type: (...) -> None
+        """Deletes SIP trunk.
+
+        :param trunk_fqdn: FQDN of the trunk to be deleted.
+        :type trunk_fqdn: str
+        :returns: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+        """
+        if trunk_fqdn is None:
+            raise ValueError("Parameter 'trunk_fqdn' must not be None.")
+
+        self._rest_service.patch_sip_configuration(
+            body=SipConfiguration(trunks={trunk_fqdn:None}),
+            **kwargs)
+
+    @distributed_trace
+    def get_trunks(
+        self,
+        **kwargs  # type: Any
+    ):  # type: (...) -> List[SipTrunk]
+        """Getter for currently configured SIP trunks.
```

Comment:
'Getter' sounds very Java ;)
Maybe just 'Retrieves an iterable of currently configured SIP trunks."?

Reply:
Changed to iterables.